src/queue.rs

```diff
@@ -56,14 +56,15 @@ pub struct SourcesQueueInput {
 impl SourcesQueueInput {
     /// Adds a new source to the end of the queue.
     #[inline]
-    pub fn append<T>(&self, source: T)
+    pub fn append<T>(&self, source: T) -> usize
```

Comment:
What's the rationale for changing this to return a `usize`? By analogy, [`Vec::append`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.append) doesn't return a value either.

Reply:
Oh haha, I needed this for some tests in zed. I'll throw it out. Good catch!